### PR TITLE
[nodejs] Add Error.code property

### PR DESCRIPTION
**Pull request by sosoba**
Add Error.code property

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [https://nodejs.org/dist/latest-v16.x/docs/api/errors.html#errorcode](https://nodejs.org/dist/latest-v16.x/docs/api/errors.html#errorcode)

Solution for https://github.com/DefinitelyTyped/DefinitelyTyped/issues/59692

### CODE REVIEW

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -13,6 +13,13 @@ interface ErrorConstructor {
     stackTraceLimit: number;
 }
 
+interface Error {
+  /**
+   * @see https://nodejs.org/dist/latest/docs/api/errors.html#nodejs-error-codes
+   */
+  code?: string | number;
```
Comment (LinusU) [inline types/node/globals.d.ts:20]: `code` is documented as only string here, can it really be a `number`?

ref: https://nodejs.org/dist/latest/docs/api/errors.html#errorcode_1

Comment (sosoba) [reply types/node/globals.d.ts:20]: Yes. The subprocess module set a numeric exit status here:

```js
import {it} from 'node:test';
import {rejects} from 'node:assert/strict';
import {stat} from 'node:fs/promises';
import {exec} from 'node:child_process';
import {promisify} from 'node:util';

it('Error with string code', async() => rejects(
  async() => stat('unexistsFile'), 
  e => e instanceof Error && typeof e.code === 'string'),
);

const asyncExec = promisify(exec);

it('Error with number code', async() => rejects(
  async() => asyncExec('unexistsCommand'),
  e => e instanceof Error && typeof e.code === 'number'),
);
```

**Review by SimonSchick — CHANGES_REQUESTED** [review]
We generally had been using `NodeJS.ErrnoException` for this, defining it on all errors is misleading imo.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request closed without merging**